src/header/js/header.js

```diff
@@ -415,5 +422,13 @@ export default (data) => {
 		}
 	});
 
+	$header.addItemsToCart = (count) => {
```

Comment:
This is kind of weird, we don't have imperative methods on the header for any other feature...

Reply:
yea I couldn't think of a better way to expose the method - open to ideas

Reply:
Just have clients emit events on the header, you don't really need these at all, all they do is dispatch an event, which the consuming code could do

Reply:
@dlewis-esri I agree with @paulcpederson here. You can update all components with a custom event. It's definitely hard to wrap your head around at first. Here's some handy code I use in the home app to help me out with this.

```js
    // Here we update the account component]
   // -- Make sure you're targeting the correct  dom node (2nd arg)
    updateComponent(
      "header:update:account",
      document.querySelector(".esri-header-client").lastChild, 
      updatedStructure
    );
    // I created this helper function to abstract away the custom event logic
    function updateComponent(componentEvent, componentElement, componentData) {
      var event = document.createEvent("CustomEvent");
      event.initCustomEvent(componentEvent, true, true, componentData);
      componentElement.dispatchEvent(event);
    }
```